internal/pkg/client/client.go

```diff
@@ -176,10 +192,15 @@ func (uc *client) Apply(c context.Context, obj runtime.Object) error {
 		return err
 	}
 	i, err := r.Get(u.GetName(), metav1.GetOptions{})
-	if err != nil {
+
+	if err != nil && !errors.IsNotFound(err) {
 		return err
 	}
 
+	if errors.IsNotFound(err) {
+		return uc.Create(c, obj, nil)
+	}
+
```

Comment:
@pwittrock 

Iam doing a simple json merge patch.
Looked at strategicMergePatch option but the API requires a concrete type. Would it work with unstructured ?

https://github.com/kubernetes/apimachinery/blob/def12e63c512da17043b4f0293f52d1006603d9f/pkg/util/strategicpatch/patch.go#L94

How do we generate strategicMergePatch without a concrete type ?

Reply:
For unstructured, metacontroller has implemented its own patching: 
https://github.com/GoogleCloudPlatform/metacontroller/blob/f6021f85a97b44c202678658022400a78212b4ea/dynamic/apply/apply.go

Reply:
SMP should work against unstructured.  It *should* correctly merge the types it knows about, and make default decisions for types it doesn't know about.